style: 2021-09-20 linter warnings batch 19 / 26

aldousalvarez opened the pull request:
Fixes #1368

petermetz commented on packages/cactus-plugin-ledger-connector-quorum/src/test/typescript/integration/plugin-ledger-connector-quorum/deploy-contract/openapi/openapi-validation.test.ts:
```diff
@@ -359,20 +360,21 @@ test(testCase, async (t: Test) => {
         `Endpoint ${fRun} without required transactionConfig: response.status === 400 OK`,
       );
-      const fields = e.response.data.map((param: any) =>
+      const fields = e?.response?.data.map((param) =>
```
@aldousalvarez

Casting the whole exception is also a good solution, but in the interest of keeing the diff smaller on account of conflicts that I foresee with other PRs, please fix the linter warning with a one liner like this: 

```typescript
const fields = e.response.data.map((param: { path: string }) =>
```

petermetz commented on packages/cactus-plugin-ledger-connector-quorum/src/test/typescript/integration/plugin-ledger-connector-quorum/deploy-contract/openapi/openapi-validation.test.ts:
```diff
@@ -394,17 +396,18 @@ test(testCase, async (t: Test) => {
         `Endpoint ${fRun} with fake=4: response.status === 400 OK`,
       );
-      const fields = e.response.data.map((param: any) =>
+      const fields = e?.response?.data.map((param) =>
```
@aldousalvarez

Casting the whole exception is also a good solution, but in the interest of keeing the diff smaller on account of conflicts that I foresee with other PRs, please fix the linter warning with a one liner like this: 

```typescript
const fields = e.response.data.map((param: { path: string }) =>
```

petermetz commented on packages/cactus-plugin-ledger-connector-xdai/src/test/typescript/integration/openapi/openapi-validation.test.ts:
```diff
@@ -163,19 +164,20 @@ test(testCase, async (t: Test) => {
         `Endpoint ${fDeploy} without required keychainId: response.status === 400 OK`,
       );
-      const fields = e.response.data.map((param: any) =>
+      const fields = e?.response?.data.map((param) =>
```
@aldousalvarez

Casting the whole exception is also a good solution, but in the interest of keeing the diff smaller on account of conflicts that I foresee with other PRs, please fix the linter warning with a one liner like this: 

```typescript
const fields = e.response.data.map((param: { path: string }) =>
```

petermetz commented on packages/cactus-plugin-ledger-connector-xdai/src/test/typescript/integration/openapi/openapi-validation.test.ts:
```diff
@@ -198,19 +200,20 @@ test(testCase, async (t: Test) => {
         `Endpoint ${fDeploy} with fake=4: response.status === 400 OK`,
       );
-      const fields = e.response.data.map((param: any) =>
+      const fields = e?.response?.data.map((param) =>
```
@aldousalvarez

Casting the whole exception is also a good solution, but in the interest of keeing the diff smaller on account of conflicts that I foresee with other PRs, please fix the linter warning with a one liner like this: 

```typescript
const fields = e.response.data.map((param: { path: string }) =>
```

petermetz commented on packages/cactus-plugin-ledger-connector-xdai/src/test/typescript/integration/openapi/openapi-validation.test.ts:
```diff
@@ -259,7 +262,7 @@ test(testCase, async (t: Test) => {
         },
       };
       await apiClient.invokeContractV1(
```
@aldousalvarez The original casting to `any` here was not necessary to begin with so you can avoid the entire ordeal by just doing this that also fixes the linter warning and makes it simpler to read:

```typescript
await apiClient.runTransactionV1(parameters);
```

petermetz reviewed (commented):
@aldousalvarez Is this ready for review? If yes please use the re-request review button.

aldousalvarez commented:
@petermetz Yes it is ready for review. I have re-requested review already. Sorry for the delay

petermetz commented on packages/cactus-plugin-ledger-connector-xdai/src/test/typescript/integration/openapi/openapi-validation.test.ts:
```diff
@@ -163,15 +163,15 @@ test(testCase, async (t: Test) => {
       };
       await apiClient.deployContractV1(
-        (parameters as any) as DeployContractV1Request,
+        (parameters as unknown) as DeployContractV1Request,
```
@aldousalvarez Thanks for the updates so far. I missed this one earlier, sorry. Same deal here as with the other API call: The cast should not be necessary at all because the correct input object has to be used anyway, so please update this the same way as you did with the other one.

petermetz reviewed (changes requested):
@aldousalvarez Thank you for the updates. You still need to fix the DCO sign-off as well. Instructions are behind the 'details' link of the check that is failing on the bottom of the PR page.

aldousalvarez commented:
@petermetz Okay I will update all the necessary changes needed. Thank you for the feedback will fix also the DCO sign off as well.

aldousalvarez commented:
Hello @petermetz I already fixed the changes that you requested and fixed the DCO sign off as well. I am sorry for my mistakes of the other commits and mistakes that were added. Will keep on improving my git mechanics. Thank you so much for your understanding.

petermetz reviewed (changes requested):
@petermetz Okay I will update all the necessary changes needed. Thank you for the feedback will fix also the DCO sign off as well.

@aldousalvarez No worries at all, git is **hard** :-) Not sure what you did exactly but this is not the intended result unfortunately: You still have 30 commits and the diff has 272 files in it instead of just the few files that you actually made changes on.
At this point what you want to do is rebase interactively and then drop commits from your branch that were not made by you so that you are left with a single commit (the one that you made yourself) and that should reduce the diff back to it's expected size.

aldousalvarez commented:
Hi @petermetz already fixed the non related commits and files. Thanks for the rebase interactively command it worked and it helped fixed the problem.

aldousalvarez commented:
Hello @petermetz I just squashed my commits into one. Is this correct output?

aldousalvarez commented:
Hello @petermetz done squash + rebase and force push with lease. Is this the correct output that should happen?

petermetz reviewed (approved):
@aldousalvarez Glad to hear that and you are most welcome! Thank you for the fixes, LGTM!

izuru0 reviewed (approved):
LGTM

petermetz commented on packages/cactus-plugin-ledger-connector-xdai/src/test/typescript/integration/openapi/openapi-validation.test.ts:
```diff
@@ -247,20 +243,19 @@ test(testCase, async (t: Test) => {
+        contractName: "",
         keychainId: keychainPlugin.getKeychainId(),
         invocationType: EthContractInvocationType.Call,
         methodName: "sayHello",
         params: [],
         gas: 1000000,
-        signingCredential: {
+        web3SigningCredential: {
           ethAccount: whalePubKey,
           secret: whalePrivKey,
           type: Web3SigningCredentialType.PrivateKeyHex,
         },
       };
-      await apiClient.invokeContractV1(
-        (parameters as any) as InvokeContractV1Request,
-      );
+      await apiClient.invokeContractV1(parameters);
```
@aldousalvarez Sorry, my bad. In this particular case the casting was necessary because of the nature of the assertion itself (negative case where we are knowingly sending in invalid input and the compiler must be told explicitly to leave us alone).
Here's the suggestion of mine on how to refactor it so that there is no linter warning but the test is still passing (right now it is failing in the CI):

```suggestion
      const parameters = {
        keychainId: keychainPlugin.getKeychainId(),
        invocationType: EthContractInvocationType.Call,
        methodName: "sayHello",
        params: [],
        gas: 1000000,
        web3SigningCredential: {
          ethAccount: whalePubKey,
          secret: whalePrivKey,
          type: Web3SigningCredentialType.PrivateKeyHex,
        },
      };
      // Casting here is necessary because we are knowingly sending in an
      // incorrect request to verify that the validation catches it.
      await apiClient.invokeContractV1(
        (parameters as unknown) as InvokeContractV1Request,
      );
```

petermetz commented on packages/cactus-plugin-ledger-connector-xdai/src/test/typescript/integration/openapi/openapi-validation.test.ts:
```diff
@@ -153,6 +152,7 @@ test(testCase, async (t: Test) => {
   test(`${testCase} - ${fDeploy} - ${cWithoutParams}`, async (t2: Test) => {
     try {
       const parameters = {
+        keychainId: "",
```
@aldousalvarez Sorry, I made a mistake, this is a negative test case and the casting is necessary. You can get rid of the linter warning produced by the `any` if you instead cast to `unknown` first like you did in the beginning.

```suggestion
      const parameters = {
        contractName: HelloWorldContractJson.contractName,
        constructorArgs: [],
        web3SigningCredential: {
          ethAccount: whalePubKey,
          secret: whalePrivKey,
          type: Web3SigningCredentialType.PrivateKeyHex,
        },
        gas: 1000000,
      };
      await apiClient.deployContractV1(
        (parameters as unknown) as DeployContractV1Request,
      );
```

petermetz reviewed (changes requested):
@aldousalvarez Apologies I mislead you with my earlier suggestions about a couple of the casts being not necessary (they are). Please see the comments inline and once updated then it's LGTM

aldousalvarez commented:
Hello @petermetz Just finished working on this issue and fixed all the changes that you have requested. It is okay it also helps me to apply and practice my git mechanics as well. Thank you

petermetz commented:
> Hello @petermetz Just finished working on this issue and fixed all the changes that you have requested. It is okay it also helps me to apply and practice my git mechanics as well. Thank you

@aldousalvarez Great, thank you and sorry for the back and forth!

petermetz reviewed (approved):
LGTM

aldousalvarez commented:
Hello @petermetz, I am sorry I just fixed the duplicated commit message.

petermetz commented:
> Hello @petermetz, I am sorry I just fixed the duplicated commit message.

@aldousalvarez No worries at all and thank you for the fix!
